test/governance/extensions/GovernorStorage.test.js

```diff
@@ -101,7 +101,7 @@ describe('GovernorStorage', function () {
           expect(await this.mock.proposalCount()).to.equal(1n);
 
           expect(await this.mock.proposalDetailsAt(0n)).to.deep.equal([
-            this.proposal.id,
+            await this.proposal.id,
```

Comment:
AFAIK, the mock used here is not using sequential id. Therefore, `this.proposal.id` is not returning a promise, so there is no need to await it.

Reply:
```suggestion
            this.proposal.id,
```

Reply:
The test fails without the await. I'm not sure why but it seems like within the chai deep equ it doesn't resolve without await.

Reply:
I figured it out. `GovernorStorage` contains a `proposalCount()` function. This is seen by the helper as a sign that we are using the `GovernorSequentialProposalId` module when we are not. So the issue is in the helper hability to detect `GovernorSequentialProposalId`

Reply:
Now that `GovernorStorage` is mentioned, I took a look at how these extensions would work together and it would be a bit odd (not necessarily a security issue though). Should we try to make them more compatible or that not necessary? An example of an oddity is that the end developer would have to decide which `proposalCount` to expose which would not be equal if the set proposal count feature is used.

Reply:
> An example of an oddity is that the end developer would have to decide which `proposalCount` to expose which would not be equal if the set proposal count feature is used.

Yes this is a problem...

Reply:
Honestly the easiest thing would be to rename `proposalCount` in this PR so it doesn't conflict. For example, `nextProposalId`.

Reply:
In both cases `proposalCount` is here to replicate GovernorBravo's interface (that includes `proposalCount`)

Reply:
Renaming the function to `nextProposalId` would actually be more accurate than `proposalCount` in this situation given that the proposal id can be offset initially. I think diverging from the `GovernorBravo` interface is fine here.